Pages/ProfilePage.cs

```diff
@@ -0,0 +1,307 @@
+        public static By IconPencilLastBy = By.XPath("//table[1]/tbody[last()]//i[@class='outline write icon']");
+        public static By IconRemoveLastBy = By.XPath("//table[1]/tbody[last()]//i[@class='remove icon']");
+        public static By IconPencilLastSkillBy = By.XPath("//div[@data-tab='second']//table/tbody[last()]//i[@class='outline write icon']");
+        public static By IconRemoveLastSkillBy = By.XPath("//div[@data-tab='second']//table/tbody[last()]//i[@class='remove icon']");
+        public static By FieldLanguageBy = By.XPath("//input[@placeholder='Add Language']");
+        public static By ChooseLanguageBy = By.XPath("//select[@class='ui dropdown']");
+        public static By FieldLanguageLastBy = By.XPath("//table[1]/tbody[last()]//input[@placeholder='Add Language']");
+        public static By ChooseLanguageLastBy = By.XPath("//table[1]/tbody[last()]//select[@class='ui dropdown']");
+        public static By FieldSkillBy = By.XPath("//input[@placeholder='Add Skill']");
+        public static By ChooseSkillBy = By.XPath("//select[@name='level']");
+        public static By BtnUpdateBy = By.XPath("//input[@value='Update']");
+        public static By LanguageOptionsBy = By.XPath("//select[@class='ui dropdown']/option");
+        public static By SkillOptionsBy = By.XPath("//div[@data-tab='second']//select[@name='level']/option");
+        public static By LanguageRowsBy = By.XPath("//div[@data-tab='first']//table/tbody");
+        public static By SkillsRowsBy = By.XPath("//div[@data-tab='second']//table/tbody");
+        public static By ActiveTabBy = By.XPath("//a[@class='item active']");
+
+        // Texts
+        public string[] LangDrop = { "Choose Language Level", "Basic", "Conversational", "Fluent", "Native/Bilingual" };
+        public string[] SkillDrop = { "Choose Skill Level", "Beginner", "Intermediate", "Expert" };
+        public IDictionary<int, string> tabIndexNames = new Dictionary<int, string>(){
+            {1, "first"},
+            {2, "second" } };
+        public string popupSuccessColor = "rgba(43, 60, 97, 1)";
+
+        // Methods
+        public void ClickSkillsTab()
+        {
+            WaitUtil.WaitVisible(_driver, TabSkillsBy).Click();
+        }
+
+        public void ClickLanguageTab()
+        {
+            _driver.FindElement(TabLanguagesBy).Click();
+        }
+
+        public void ClickAddNew()
+        {
+            _driver.FindElement(BtnAddNewLanguageBy).Click();
+        }
+
+        public bool IsElementPresent(By by)
+        {
+            try
+            {
+                _driver.FindElement(by);
+                return true;
+            }
+            catch (NoSuchElementException)
+            {
+                return false;
+            }
+        }
+
+        public bool isAtProfilePage()
+        {
+            return IsElementPresent(TabProfileBy);
+        }
+
+        public string SelectRandomLevel(By by, string[] dropDownList)
```

Comment:
I would not select random level, instead you should test all levels or pick a level! your test should also be asserting that your data is present please do this

Reply:
Updated. New function added. In the scenario outline, there will be assertion for added details, i.e.:
Then User is able to see Language details: <language> <level>
Random level function is retained to be use in other places.